pkgs/by-name/po/postgresqlTestHook/package.nix

```diff
@@ -1,8 +1,18 @@
 { callPackage, makeSetupHook }:
 
-makeSetupHook {
+(makeSetupHook {
   name = "postgresql-test-hook";
   passthru.tests = {
     simple = callPackage ./test.nix { };
   };
-} ./postgresql-test-hook.sh
+} ./postgresql-test-hook.sh).overrideAttrs
+  # For shared memory on Darwin.
+  # TODO: perhaps always use mmap? I assume it should be available in sandbox.
+  # https://www.postgresql.org/docs/current/runtime-config-resource.html#GUC-SHARED-MEMORY-TYPE
+  # https://www.postgresql.org/docs/current/runtime-config-resource.html#GUC-DYNAMIC-SHARED-MEMORY-TYPE
```

Comment:
> Is this related to https://github.com/NixOS/nixpkgs/issues/371242?

I don’t think so. I’m not aware of this issue, but it seems to mention that builds fail on Hydra and IIRC macOS builders do not use sandbox (though things could have changed since the last time I’ve checked that).

Forcing mmap implementation instead of POSIX/SysV shared memory might help as discussed here, but I don’t have time to test that.

Reply:
- `aarch64-darwin` host
- `sandbox = true` set in `/etc/nix/nix.conf`
- `sudo launchctl kickstart -k system/org.nixos.nix-daemon`
  to restart Nix daemon if `sandbox = true` was not there before
- **Nix 2.18.1**
- I’ve doubled-checked with Nix version 2.24.12 and this seems to be fixed https://github.com/NixOS/nix/pull/10878, but older Nix version still require an explicit sandbox profile.

```
nix build --print-build-logs --no-link --impure --expr 'with import ./. { };
stdenv.mkDerivation {
  name = "postgresql-test-hook-repro";
  dontUnpack = true;
  doCheck = true;
  nativeCheckInputs = [
    postgresqlTestHook
    postgresql
  ];
  installPhase = ''
    runHook preInstall
    mkdir -p "$out"
    runHook postInstall
  '';
}'
```